Userland/Applications/PixelPaint/ImageEditor.cpp

```diff
@@ -426,6 +427,30 @@ void ImageEditor::set_active_layer(Layer* layer)
     }
 }
 
+bool ImageEditor::add_new_layer_from_selection()
+{
+    auto current_layer_selection = image().selection();
+    VERIFY(!current_layer_selection.is_empty());
```

Comment:
This should error out instead and not crash as it is user error, not programming error. I know that you're enabling and disabling the action, but I think we can be more lenient here if that ever goes wrong.

Reply:
Yep, makes sense now. Didn't realize that VERIFY causes a crash if it fails. I've fixed this.